### PR TITLE
Add .to_string() to axis y/xlabel

### PR DESCRIPTION
**Pull request by aaguasca**
**Description**

This pull request addresses Issue #4145. 

The changes consist of adding `.to_string('latex_inline')` to the unit representation in x/y axis labels using the units in ax.xaxis.units, ax.yaxis.units, or directly the units of the object displayed in that axis. In particular, it is added to axes where a power representation is required.

**Dear reviewer**

The PR is ready for review. 

Most of the code where I add `.to_string('latex_inline')` already uses `with quantity_support()`, which prevents any possible error when the ax.yaxis.units or ax.xaxis.units are NoneType. For the case of methods `plot` and `plot_ts_profiles` in "gammapy/estimators/points/core.py", the function variable `DEFAULT_UNIT` prevents NoneType ax.yaxis.units.

### CODE REVIEW

**Comment by registerrier** [comment]
Thanks @aaguasca !
Apparently there is a formatting issue in the CI. Do you use the correct settings for the pre-commit? What version of black do you have?

**Comment by adonath** [comment]
Thanks @aaguasca, this looks mostly good! I think it is still a good idea to introduce a global variable e.g. here https://github.com/gammapy/gammapy/blob/main/gammapy/maps/axes.py#L71

Like so:
```python3

UNIT_STRING_FORMAT = "latex_inline"

ax.set_ylabel(f"Sensitivity [{t['e2dnde'].unit.to_string(UNIT_STRING_FORMAT)}]")
```
Such that it could be potentially modified by users, if needed.

**Comment by adonath** [comment]
On second thought I think we should use is consistently (not only for powers...). There potentially a few more places to be adapted https://github.com/gammapy/gammapy/search?q=set_ylabel

**Comment by registerrier** [comment]
Hi @aaguasca , there is a merge conflict here because the tests you modify were adapted in #4429 because they were breaking with future versions of astropy and matplotlib. The order of the units in the label will change in the future, so we no longer test the unit in the label string, but simply that the unit seen by matplotlib is correct.
So the modifications in test_spectral should be reverted and rebased. Then maybe simply test that the unit part of the label string is latex?

**Comment by aaguasca** [comment]
Hi @registerrier. Thank you for the clarification. I will revert the changes in the test test_spectral. 

I created a PR to fix another problem with the tests (PR #4446). Once it is merged, I will push all commits. Please, let me know if I should modify something in PR #4446.

> Then maybe simply test that the unit part of the label string is latex?

Maybe if we know the convention of the unit representation, for example, within brackets, checking the default string format representation of the units should be relatively easy using split.

**Review by registerrier — COMMENTED** [review]
Thanks @aaguasca . This looks good.

Shouldn't all `MapAxis` have a `Unit` defined? Even `LabelMapAxis` does.
For `rad`, the issue is the content of the `IRF`, but here it should have a unit too, no?

**Comment by aaguasca** [comment]
I think the problem is `psf_value`. The returned array by the method `self.psf_map.interp_by_coord` has no units because `self.psf_map` is a `RegionNDMap` object.

Maybe a workaround would be to multiply the output `psf_value` by `self.psf_map.unit`. What do you think the best solution is?

File: gammapy/irf/psf/map.py [hunk 1]
```diff
@@ -493,9 +497,10 @@ def plot_psf_vs_rad(self, ax=None, energy_true=[0.1, 1, 10] * u.TeV, **kwargs):
             with quantity_support():
                 ax.plot(rad, psf_value, label=label, **kwargs)
 
+        format_NoneType_axes_units(ax)
```
Comment (registerrier) [inline gammapy/irf/psf/map.py:500]: yes here it should  be sufficient to do `psf_value *= self.psf_map.unit`

**Review by registerrier — APPROVED** [review]
Thansk @aaguasca . This looks good. Indeed, you could multiply here the `psf_value` by the `psf_map` unit. 
`interp_by_coord` returns an array...

**Review by registerrier — APPROVED** [review]
Thanks @aaguasca . No further comment.

**Pull request merged**